File: ccif/src/java/other/org/himalay/ccif/cblurlmsg.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class cblurlmsg extends BinStruct implements PublicBinMsg {




    // members variables
    // urlType
    public short urlType ;
    // urlchars
    public ByteArray urlchars ;

    public cblurlmsg () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize urlType

        // Initialize urlchars
        urlchars= new ByteArray();
        urlchars.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read urlType
        {
            urlType=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read urlchars
        {
            retVal+=urlchars.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;


        {   /** fix dependent sizes for urlchars**/
        }

        // write urlType
        ostream.writeByte(urlType);
        retVal +=1;
        // write urlchars
        {
            retVal += urlchars.write(ostream);
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("cblurlmsg\n");
        dc.increaseIndent();
        int retVal= 0;
        // write urlType
        dc.indent();
        dc.getPs().println("urlType="+urlType+"(0x"+ Integer.toHexString(urlType) + ")" );
        // write urlchars
        dc.indent();
        dc.getPs().print("urlchars: "+urlchars.getSize()+"(0x"+Integer.toHexString(urlchars.getSize())+")\n");
        this.urlchars.dump(dc);
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for urlType
    //public short getUrlType()
    //{
    //    return urlType ;
    //}


    // Setter for urlType
    //public void setUrlType(short val)
    //{
    //    this.urlType= val;
    //}
    // Getter for urlchars
    //public ByteArray getUrlchars()
    //{
    //    return urlchars ;
    //}


    // Setter for urlchars
    //public void setUrlchars(ByteArray val)
    //{
    //    this.urlchars= val;
    //}


    public void setUrlchars(byte[] val)
    {
        this.urlchars.setData(val);
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/cblurlmsg.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "cblurlmsg.h"

#ifdef org_himalay_ccif_cblurlmsg__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
cblurlmsg::cblurlmsg() {
    references= 0;


    // urlType
    urlType= (ui8)0;
    // urlchars
    org_himalay_msgs_runtime_ByteArray_NEW(urlchars);
    urlchars->setSizeType("FIRST_UI8");
}
// Destructor
cblurlmsg::~cblurlmsg() {
    // urlType

    // urlchars
    org_himalay_msgs_runtime_ByteArray_DELETE( urlchars);
}

int cblurlmsg::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // urlType
    {
        urlType=(istream.readUI8());
        retVal+=1;
    }
    // urlchars
    {
        retVal+=urlchars->read(istream);
    }
    return retVal;
}

int cblurlmsg::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read urlType
    {
        urlType=(istream.readUI8());
        retVal+=1;
    }
    // read urlchars
    {
        retVal+=urlchars->read(istream);
    }

    return retVal;
}

int cblurlmsg::write(DataOutputStream& ostream) { // throws IOException

    {   /** fix dependent sizes for urlchars**/
    }

    int retVal= 0;
    // write urlType
    ostream.writeUI8(urlType);
    retVal +=1;
    // write urlchars
    {
        retVal += urlchars->write(ostream);
    }

    return retVal;
}

int cblurlmsg::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("cblurlmsg\n");
    dc.increaseIndent();
    int retVal= 0;
    // write urlType
    dc.indent();
    dc.getPs().print("urlType=");
    dc.getPs().println((long)urlType);
    // write urlchars
    dc.indent();
    dc.getPs().print("urlchars");
    urlchars->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for urlType
//ui8 cblurlmsg::getUrlType()
//{
//return this->urlType;
//}

// Setter for urlType
//void cblurlmsg::setUrlType(ui8 val)
//{
//this->urlType= val;
//}

// Getter for urlchars
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE cblurlmsg::getUrlchars()
//{
//return this->urlchars;
//}

// Setter for urlchars
//void cblurlmsg::setUrlchars(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->urlchars= val;
//}

int cblurlmsg::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_cblurlmsg__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::cblurlmsg* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::cblurlmsg* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code